app/cli/cmd/plugins.go

```diff
@@ -0,0 +1,398 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/chainloop-dev/chainloop/app/cli/internal/action"
+	"github.com/chainloop-dev/chainloop/app/cli/pkg/plugins"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	stringFlagType = "string"
+	boolFlagType   = "bool"
+	intFlagType    = "int"
+)
+
+var (
+	pluginManager      *plugins.Manager
+	registeredCommands map[string]string // Track which plugin registered which command
+)
+
+func init() {
+	registeredCommands = make(map[string]string)
+}
+
+func newPluginCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "plugin",
+		Short: "Manage plugins (preview)",
+		Long:  "Manage Chainloop CLI plugins (preview)",
+	}
+
+	cmd.AddCommand(newPluginListCmd())
+	cmd.AddCommand(newPluginDescribeCmd())
+	cmd.AddCommand(newPluginInstallCmd())
+
+	return cmd
+}
+
+func createPluginCommand(rootCmd *cobra.Command, plugin *plugins.LoadedPlugin, cmdInfo *plugins.PluginCommandInfo) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   cmdInfo.Name,
+		Short: cmdInfo.Description,
+		Long:  fmt.Sprintf("%s\n\nProvided by plugin: %s v%s", cmdInfo.Description, plugin.Metadata.Name, plugin.Metadata.Version),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			// Collect all flags that were set
+			flags := make(map[string]*plugins.SimpleFlag)
+
+			for _, flag := range cmdInfo.Flags {
+				simpleFlag := &plugins.SimpleFlag{
+					Name:      flag.Name,
+					Shorthand: flag.Shorthand,
+					Usage:     flag.Description,
+				}
+
+				switch flag.Type {
+				case stringFlagType:
+					if val, err := cmd.Flags().GetString(flag.Name); err == nil {
+						simpleFlag.Value = val
+					}
+				case boolFlagType:
+					if val, err := cmd.Flags().GetBool(flag.Name); err == nil {
+						simpleFlag.Value = strconv.FormatBool(val)
+					}
+				case intFlagType:
+					if val, err := cmd.Flags().GetInt(flag.Name); err == nil {
+						simpleFlag.Value = strconv.Itoa(val)
+					}
+				}
+				flags[flag.Name] = simpleFlag
+
+			}
+
+			// Collect all persistent flags that were set
+			rootCmd.PersistentFlags().VisitAll(func(f *pflag.Flag) {
+				flags[f.Name] = &plugins.SimpleFlag{
+					Name:        f.Name,
+					Shorthand:   f.Shorthand,
+					Usage:       f.Usage,
+					Value:       f.Value.String(),
```

Comment:
`f.Value.String()` is this correct? what if it's an array of objects? It's ok if we do not support it but we should at least control it and fail if we receive a flag of a type that we do not support, I think